### PR TITLE
Migrate run_desktop_tests.py to Python3 

### PR DESCRIPTION
**Pull request by strump**
Python2 is not installed on latests MacOS and is deprecated since 2020.

Made changes to `run_desktop_tests.py`, `testserver.py` and `SiblingKiller.py`

### CODE REVIEW

**Comment by strump** [comment]
Is it possible to trigger all Github checks for this PR?

**Comment by biodranik** [comment]
> Is it possible to trigger all Github checks for this PR?

There are no checks yet for these python scripts. If it's easy, you can add them :)

File: tools/python/SiblingKiller.py [hunk 1]
```diff
@@ -4,7 +4,8 @@
```
Comment (biodranik) [inline tools/python/SiblingKiller.py:4]: Does it make sense to add `#!/usr/bin/env python3` at the top?

File: tools/python/testserver.py [hunk 2]
```diff
@@ -171,7 +171,7 @@ def dispatch_response(self, payload):
             self.send_header(h, payload.headers()[h])
         self.send_header("Content-Length", payload.length())
         self.end_headers()
-        self.wfile.write(payload.message())
+        self.wfile.write(payload.message().encode('utf8'))
```
Comment (biodranik) [inline tools/python/testserver.py:174]: What is the default encoding? Why explicit encode is needed?

**Review by biodranik — APPROVED** [review]

File: tools/python/testserver.py (hunk 2 above)
Comment (strump) [reply tools/python/testserver.py:174]: `payload.message()` method returns string, while `write` method expects bytes object. Call to `encode` is needed to convert string -> bytes.

**Comment by dbf256** [comment]
> Is it possible to trigger all Github checks for this PR?

You can trigger checks at your fork of this repo.

File: tools/python/SiblingKiller.py (hunk 1 above)
Comment (strump) [reply tools/python/SiblingKiller.py:4]: In [`INSTALL.md`](https://github.com/organicmaps/organicmaps/blob/master/docs/INSTALL.md) guide there's example how to run tests:

```
../tools/python/run_desktop_tests.py -f . -u ../data/ -d ../data/
```

If remove the first line from `run_desktop_tests.py` then this command will not work.

Comment (biodranik) [reply tools/python/SiblingKiller.py:4]: It's not about removing the hashbang from run_desktop_tests.py, it's about adding it into SiblingKiller.py

Comment (strump) [reply tools/python/SiblingKiller.py:4]: Well, `SiblingKiller.py` should not be run from command line. Only as Python module. So we don’t need `#!/usr/..` as the first line.

**Comment by biodranik** [comment]
@strump looks like there are some issues with these changes. Can you please carefully check why HTTP server fails in tests? ASSERT fails, because Cancel doesn't work properly.
```
$ tools/python/run_desktop_tests.py -f ../omim-build-debug -i platform_tests

/Users/alex/Developer/omim/omim/tools/python/run_desktop_tests.py:83: DeprecationWarning: The 'warn' function is deprecated, use 'warning' instead
  logging.warn("-i or -b option found, the -e option will be ignored")
2022-07-15 08:41:20,298 -i or -b option found, the -e option will be ignored
2022-07-15 08:41:20,298 Init server. Pid: 61432
2022-07-15 08:41:20,298 >> ps -f
2022-07-15 08:41:20,307 Sibling killer: my process id = 61432
2022-07-15 08:41:20,314 Pinging returned html: None
2022-07-15 08:41:20,315 Pinging returned html: None
2022-07-15 08:41:20,315 There are no servers that are currently serving. Will try to kill our siblings.
2022-07-15 08:41:20,315 >> lsof -a -p932 -i4
2022-07-15 08:41:20,674 >> lsof -a -p61432 -i4
2022-07-15 08:41:21,029 >> lsof -a -p936 -i4
2022-07-15 08:41:21,376 >> lsof -a -p933 -i4
2022-07-15 08:41:21,723 >> lsof -a -p934 -i4
2022-07-15 08:41:22,071 >> lsof -a -p36060 -i4
2022-07-15 08:41:22,422 Pinging returned html: None
2022-07-15 08:41:22,423 Pinging returned html: None
2022-07-15 08:41:22,423 There are no siblings to kill

*************
WARNING: Using the python's built-in BaseHTTPServer!
It is all right if you run the tests on your local machine, but if you are running tests on a server,
please consider installing Tornado. It is a much more powerful web-server. Otherwise you will find
that some of your downloader tests either fail or hang.

do

sudo pip install tornado

or go to http://www.tornadoweb.org/en/stable/ for more detail.
*************

2022-07-15 08:41:22,424 Started server with pid: 61432
2022-07-15 08:41:25,425 platform_tests
2022-07-15 08:41:25,431 Pid: 61449
Running country_file_tests.cpp::CountryFile_Smoke
OK
Test took 0 ms

Running downloader_test.cpp::DownloaderSimpleGet
ID: 105553137891200 Starting data task for http://localhost:34568/unit_tests/1.txt
127.0.0.1 - - [15/Jul/2022 08:41:25] "GET /unit_tests/1.txt HTTP/1.1" 200 -
Got response with status code 200
ID: 105553137891200 Connection is destroyed
ID: 105553137763776 Starting data task for http://localhost:34568/unit_tests/permanent
127.0.0.1 - - [15/Jul/2022 08:41:25] "GET /unit_tests/permanent HTTP/1.1" 301 -
Canceling because of redirect from http://localhost:34568/unit_tests/permanent to http://localhost:34568/unit_tests/google.com
HttpRequest error: 301
ID: 105553137763776 Connection is destroyed
ID: 105553137765632 Starting data task for http://localhost:34568/unit_tests/notexisting_unittest
127.0.0.1 - - [15/Jul/2022 08:41:25] "GET /unit_tests/notexisting_unittest HTTP/1.1" 404 -
Got response with status code 404
Received invalid HTTP status code, canceling download 404
HttpRequest error: 404
ID: 105553137765632 Connection is destroyed
ID: 105553137839936 Starting data task for http://not-valid-host123532.ath.cx
HttpRequest error: -1003
ID: 105553137839936 Connection is destroyed
ID: 105553137895872 Starting data task for http://localhost:34568/unit_tests/47kb.file
127.0.0.1 - - [15/Jul/2022 08:41:25] "GET /unit_tests/47kb.file HTTP/1.1" 404 -
Got response with status code 404
Received invalid HTTP status code, canceling download 404
HttpRequest error: 404
FAILED
downloader_tests/downloader_test.cpp:119 TEST(false) Should be never called
libc++abi: terminating with uncaught exception of type TestFailureException: TestFailureException /Users/alex/Developer/omim/omim/testing/testing.hpp:41, "downloader_tests/downloader_test.cpp 119 TEST(false) Should be never called"
2022-07-15 08:41:25,735 Kill called in ResponseProvider
2022-07-15 08:41:25,735 Kill called in testserver
127.0.0.1 - - [15/Jul/2022 08:41:25] "GET /kill HTTP/1.1" 200 -
2022-07-15 08:41:25,736
PASSED
2022-07-15 08:41:25,736 - platform_tests
2022-07-15 08:41:26,238 The server's life has come to an end, pid: 61432
```

**Pull request merged**